endpoints/openrtb2/test_utils.go

```diff
@@ -1190,7 +1190,13 @@ func (te *exchangeTestWrapper) HoldAuction(ctx context.Context, r *exchange.Auct
 	}
 
 	// Save the validated bidRequest that we are about to feed HoldAuction
-	te.actualValidatedBidReq = r.BidRequestWrapper.BidRequest
+	if r.BidRequestWrapper.BidRequest != nil {
+		v, err := jsonutil.Marshal(r.BidRequestWrapper.BidRequest)
+		if err != nil {
+			return nil, fmt.Errorf("Error converting actual validated bid request to json: %s", err)
+		}
+		te.actualValidatedBidReq = v
```

Comment:
Is the `exchangeTestWrapper` reused? if so then you will need to nil out the actual bid request when one does not exist, otherwise it will be the same as the last test.

Reply:
It isn't reused. A single instance of it is created for each test case in a test.